genesyscloud/telephony_providers_edges_phonebasesettings/resource_genesyscloud_telephony_providers_edges_phonebasesettings.go

```diff
@@ -180,39 +179,21 @@ func deletePhoneBaseSettings(ctx context.Context, d *schema.ResourceData, meta i
 			return nil
 		}
 
-		return retry.RetryableError(fmt.Errorf("Phone base settings %s still exists", d.Id()))
+		return retry.RetryableError(fmt.Errorf("phone base settings %s still exists", d.Id()))
 	})
 }
 
 func getAllPhoneBaseSettings(ctx context.Context, sdkConfig *platformclientv2.Configuration) (resourceExporter.ResourceIDMetaMap, diag.Diagnostics) {
 	resources := make(resourceExporter.ResourceIDMetaMap)
+	pp := getPhoneBaseProxy(sdkConfig)
+	phoneBaseSettings, err := pp.getAllPhoneBaseSettings(ctx)
+	if err != nil {
+		return nil, diag.Errorf("failed to get all phone base settings: %s", err)
+	}
 
-	edgesAPI := platformclientv2.NewTelephonyProvidersEdgeApiWithConfig(sdkConfig)
-
-	err := gcloud.WithRetries(ctx, 15*time.Second, func() *retry.RetryError {
-		for pageNum := 1; ; pageNum++ {
-			const pageSize = 100
-			phoneBaseSettings, resp, getErr := edgesAPI.GetTelephonyProvidersEdgesPhonebasesettings(pageSize, pageNum, "", "", nil, "")
-			if getErr != nil {
-				if gcloud.IsStatus404(resp) {
-					return retry.RetryableError(fmt.Errorf("Failed to get page of phonebasesettings: %v", getErr))
-				}
-				return retry.NonRetryableError(fmt.Errorf("Failed to get page of phonebasesettings: %v", getErr))
-			}
-
-			if phoneBaseSettings.Entities == nil || len(*phoneBaseSettings.Entities) == 0 {
-				break
-			}
-
-			for _, phoneBaseSetting := range *phoneBaseSettings.Entities {
-				if phoneBaseSetting.State != nil && *phoneBaseSetting.State != "deleted" {
-					resources[*phoneBaseSetting.Id] = &resourceExporter.ResourceMeta{Name: *phoneBaseSetting.Name}
-				}
-			}
-		}
-
-		return nil
-	})
+	for _, phoneBaseSetting := range *phoneBaseSettings {
```

Comment:
it might result in a runtime panic due to a nil pointer dereference if *phoneBaseSettings is nil
which could be a possibility with declaration of 
var allPhoneBaseSettings []platformclientv2.Phonebase in the proxy class

Reply:
adding an if statement to check for nil before dereferencing it.

Reply:
addressed